src/modules/land_detector/MulticopterLandDetector.cpp

```diff
@@ -209,20 +210,21 @@ bool MulticopterLandDetector::_get_ground_contact_state()
 	// low thrust: 30% of throttle range between min and hover, relaxed to 60% if hover thrust estimate available
 	const float thr_pct_hover = _hover_thrust_estimate_valid ? 0.6f : 0.3f;
 	const float sys_low_throttle = _params.minThrottle + (_params.hoverThrottle - _params.minThrottle) * thr_pct_hover;
-	bool ground_contact = (_actuator_controls_throttle <= sys_low_throttle);
+	_has_low_throttle = (_actuator_controls_throttle <= sys_low_throttle);
+	bool ground_contact = _has_low_throttle;
 
 	// if we have a valid velocity setpoint and the vehicle is demanded to go down but no vertical movement present,
 	// we then can assume that the vehicle hit ground
 	if (_flag_control_climb_rate_enabled) {
-		vehicle_local_position_setpoint_s vehicle_local_position_setpoint;
+		vehicle_local_position_setpoint_s trejactory_setpoint;
 
-		if (_vehicle_local_position_setpoint_sub.update(&vehicle_local_position_setpoint)) {
+		if (_trajectory_setpoint_sub.update(&trejactory_setpoint)) {
 			// setpoints can briefly be NAN to signal resets, TODO: fix in multicopter position controller
-			const bool descend_vel_sp = PX4_ISFINITE(vehicle_local_position_setpoint.vz)
-						    && (vehicle_local_position_setpoint.vz >= land_speed_threshold);
+			const bool descend_vel_sp = PX4_ISFINITE(trejactory_setpoint.vz)
+						    && (trejactory_setpoint.vz >= land_speed_threshold);
 
-			const bool descend_acc_sp = PX4_ISFINITE(vehicle_local_position_setpoint.acceleration[2])
-						    && (vehicle_local_position_setpoint.acceleration[2] >= 100.f);
+			const bool descend_acc_sp = PX4_ISFINITE(trejactory_setpoint.acceleration[2])
+						    && (trejactory_setpoint.acceleration[2] >= 100.f);
```

Comment:
I know that one isn't new, but quick comment on where the 100 here are coming from would help.

Reply:
It comes from the position controller during ground contact, so this probably isn't right in trajectory_setpoint.

https://github.com/PX4/PX4-Autopilot/blob/59b70a881d5d5eed93ebe97889a1ae7886a9520a/src/modules/mc_pos_control/MulticopterPositionControl.cpp#L373-L376

Reply:
Yes, I left it in because there's an in-between branch that has the takeoff state machine in the flight mode manager and still replaces the setpoints up on ground contact. On master, this is not necessary anymore with the new takeoff state machine location and looking at the trajectory setpoint here.

Reply:
I removed it since it's not very useful on master anymore:
https://github.com/PX4/PX4-Autopilot/pull/17440/commits/90139c680c2bad342e539678134a4fb8b492796c#diff-ccf6ef052db2503b2701a1211d31f99d45cf581db98e056ca8b4da725a805b7dL220-L221